### ISSUE TITLE
project 3d pts onto 2d image plane

### ISSUE THREAD

**Issue by muses0229** [boycehbz/CHOMP#3]
Hi,now I am trying to project world 3d pts onto 2d image plane,and the results doesn't look right.I am not familiar with the OcMotion dataset,could you please give me some advices?

**Comment by yejr0229** [comment]
![22](https://github.com/boycehbz/CHOMP/assets/132911839/98d44825-ff4b-47b8-9630-1617c7e71c27)

**Comment by yejr0229** [comment]
Here is my code to project points,the R,T,K are all cameras' parameters.
RT = torch.cat([torch.tensor(data.R.transpose()), torch.tensor(data.T).reshape(3,1)], -1)[None, None].cuda()
xyz = torch.repeat_interleave(world_3d_pts.squeeze(0)[None, None], repeats=RT.shape[1], dim=1)
xyz = torch.matmul(RT[:, :, None, :, :3].float(), xyz[..., None].float()) + RT[:, :, None, :, 3:].float()
xyz = torch.matmul(torch.tensor(data.K)[None, None][:, :, None].float().cuda(), xyz)[..., 0]
xy = xyz[..., :2] / (xyz[..., 2:] + 1e-5)
src_uv = xy.view(-1, *xy.shape[2:]).squeeze(0) # (6890,2)

**Comment by boycehbz** [comment]
It seems your projection code is wrong. You may refer to  Line 63 in [this file](https://github.com/boycehbz/GroupRec/blob/main/utils/geometry.py)

**Comment by muses0229** [comment]
The result is still not right..., I'd like to know is R,T in smpl coordinate or world coordinate?

**Comment by yejr0229** [comment]
Here is the projected image I get now:
![bb](https://github.com/boycehbz/CHOMP/assets/132911839/5299190f-e06f-4fd0-a37c-3f2b1b6a2353)

**Comment by boycehbz** [comment]
I have added projection code and you can use it by setting is_render = False in Line 92 https://github.com/boycehbz/CHOMP/blob/main/vis_OcMotion.py

**Comment by yejr0229** [comment]
sorry for bothering again,but my projection result still seems a bit wrong:
![22](https://github.com/boycehbz/CHOMP/assets/132911839/f836e621-d212-428b-9b0c-8869be0d12d7)
I'm pretty sure my projection code is right because I get the same result using your code.
Here is my code to transform points from canonical space to observation space,I remove 'Th' and get the above result but still not right.
![DZNY81JTI`2}7R}0QQ@@B_U](https://github.com/boycehbz/CHOMP/assets/132911839/88e2aeb3-61a1-4556-8280-a7e2c21df937)

**Comment by yejr0229** [comment]
I check my code and found I add wrong transpose on camera.R,now my projection result is like that:
![F2N%KB SG13W$JUTBBEK~5L](https://github.com/tiangexiang/OccNeRF/assets/132911839/33a19623-4a5e-4b96-98ec-0b1ca1049de5)
Is this looks right now?
